src/ir/comp.rs

```diff
@@ -668,8 +668,11 @@ impl CompInfo {
 
                     // NB: This gets us an owned `Function`, not a
                     // `FunctionSig`.
-                    let signature = match Item::parse(cur, Some(potential_id), ctx) {
-                        Ok(item) if ctx.resolve_item(item).kind().is_function() => item,
+                    let signature =
+                        match Item::parse(cur, Some(potential_id), ctx) {
+                            Ok(item) if ctx.resolve_item(item)
```

Comment:
could you leave formatting patches in a different commit? That will make this much more pleasurable to review :)